tests/sles4sap/ase_install.pm

```diff
@@ -0,0 +1,105 @@
+# SUSE's SLES4SAP openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Install SAP ASE via command line using a response file. Both
+# the installation media and the response file are supplied as openQA
+# assets. Verify installation with sles4sap/ase_test.
+#
+# This test module expects the tarball with the installation files for SAP ASE
+# in ASSET_0 and the gzipped response file in ASSET_1
+#
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+use base 'sles4sap';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use utils qw(file_content_replace);
+use Utils::Logging qw(save_and_upload_log);
+use version_utils qw(is_sle);
+
+use constant ASE_RESPONSE_FILE => '$HOME/ase_response_file.txt';
+
+=head2 prepare_system_for_ase
+
+  $self->prepare_system_for_ase
+
+Run some preparation commands in the system before SAP ASE's installation.
+This includes setting kernel.randomize_va_space to the recommended value,
+adding the hostname to /etc/hosts, creating a target directory where to
+download the ASE installer and applying the SAP-ASE profile to the
+system via saptune.
+
+=cut
+
+sub prepare_system_for_ase {
+    my ($self, %args) = @_;
+    $args{target} //= '/sapinst';
+    # Add host's IP to /etc/hosts
+    $self->add_hostname_to_hosts;
+    $self->prepare_profile('SAP-ASE');
+    # We need to ensure that the kernel parameter kernel.randomize_va_space is set to 0 in 12-SP3+
+    assert_script_run 'sysctl kernel.randomize_va_space=0' if is_sle('12-SP3+');
+    record_info 'kernel.randomize_va_space', script_output('sysctl kernel.randomize_va_space', proceed_on_failure => 1);
+    assert_script_run "mkdir -p $args{target}";
+}
+
+=head2 download_ase_assets
+
+  my $installation_dir = download_ase_assets()
+
+Download SAP ASE installation tarball (.TGZ format) from ASSET_0 setting, and dowload
+the response file require for installation (.GZ text format) from ASSET_1 setting.
+Returns the directory that contains the unpacked installation files.
+
+=cut
+
+sub download_ase_assets {
+    my %args = @_;
+    $args{target} //= '/sapinst';
+    $args{timeout} //= 600;
+    assert_script_run "pushd $args{target}";
+    # Verify ASSET_0 and ASSET_1 are set
+    get_required_var("ASSET_$_") for (0 .. 1);
+    # Download assets into SUT
+    record_info 'SAP ASE Installer', get_var('ASSET_0');
+    my $ase_location = data_url('ASSET_0');
+    assert_script_run "wget -O - $ase_location | tar -zxf -", timeout => $args{timeout};
+    # SAP ASE installer extracts its contents in a directory that begins with 'ebf'
+    # The following lines will attempt to find the directory name the files were extracted on
+    my $instdir = script_output 'echo ":$(ls -d ebf* | tail -1):"';
```

Comment:
why `tail` is needed?

Reply:
I want to prevent the case where more than one `ebf*` directory exists there. `tail -1` gives the last one, which should be the higher version. I don't expect more than one version to be there in a single test, but I'm covering for the possibility that eventually this module is scheduled in a different scenario.